### PR TITLE
Pass-through http handler during configuration

### PR DESCRIPTION
**Pull request by travispereira**
Update to pass through a http handler object during configuration to allow configuration of a proxy

### CODE REVIEW

**Comment by billpratt** [comment]
Can you provide a use case for this so I can update the docs?

File: src/Serilog.Sinks.SumoLogic/LoggerConfigurationSumoLogicExtensions.cs [hunk 1]
```diff
@@ -24,6 +25,7 @@ public static class LoggerConfigurationSumoLogicExtensions
         /// <param name="period">The time to wait between checking for event batches.</param>
         /// <param name="textFormatter">Supplies how logs should be formatted, or null to use the default</param>
         /// <param name="outputTemplate">Override default output template. Should not be used if overriding <see cref="ITextFormatter"/></param>
+        /// <param name="handler">Override default http handler <see cref="ITextFormatter"/></param>
```
Comment (billpratt) [inline src/Serilog.Sinks.SumoLogic/LoggerConfigurationSumoLogicExtensions.cs]: Change ITextFormatter to HttpMessageHandler

**Review by billpratt — CHANGES_REQUESTED** [review]

File: src/Serilog.Sinks.SumoLogic/LoggerConfigurationSumoLogicExtensions.cs (hunk 1 above)
Comment (travispereira) [reply src/Serilog.Sinks.SumoLogic/LoggerConfigurationSumoLogicExtensions.cs]: Updated

**Comment by travispereira** [comment]
The use case is to be able to access sumologic via a proxy, e.g.

```
var proxyUri = new Uri("https://host:port");
var networkCredentials = new NetworkCredential(username, password);
var proxy = new WebProxy(proxyUri, false, null, networkCredentials);

HttpClientHandler handler = new HttpClientHandler()
{
  Proxy = proxy
};

loggerConfiguration.WriteTo.SumoLogic(urlString, handler: handler);
```

**Comment by travispereira** [comment]
Hi Bill, is there anything else you need before merging this pull request?

**Review by billpratt — APPROVED** [review]
LGTM

**Comment by billpratt** [comment]
Thanks @travispereira. This change is published with v2.4.0: https://www.nuget.org/packages/Serilog.Sinks.SumoLogic/2.4.0

**Pull request merged**